src/Akka.Persistence.RavenDB/Journal/RavenDbJournal.cs

```diff
@@ -175,5 +283,59 @@ protected override async Task DeleteMessagesToAsync(string persistenceId, long t
         }
 
         private AsyncReaderWriterLock GetLocker(string persistenceId) => _lockPerActor.GetOrAdd(persistenceId, new AsyncReaderWriterLock());
+
+        public static async Task<Status> EnsureIndexesCreated(RavenDbStore store)
+        {
+            using var cts = store.GetCancellationTokenSource(false);
+            if (cts.IsCancellationRequested)
+                return new Status.Success(NotUsed.Instance);
+
+            var startTime = DateTime.Now;
+            while (DateTime.Now - TimeSpan.FromSeconds(15) < startTime)
+            {
+                try
+                {
+                    var db = await store.Instance.Maintenance.Server.SendAsync(
+                        new GetDatabaseRecordOperation(store.Configuration.Name), cts.Token);
+                    if (db == null)
+                    {
+                        Thread.Sleep(TimeSpan.FromMilliseconds(250));
+                        continue;
+                    }
+
+                    var res1 = await store.Instance.Maintenance.SendAsync(new GetIndexNamesOperation(0, int.MaxValue), cts.Token);
+                    if (res1.Contains(nameof(EventsByTagAndChangeVector)) &&
+                        res1.Contains(nameof(ActorsByChangeVector)))
+                    {
+                        return new Status.Success(NotUsed.Instance);
+                    }
+
+                    await new EventsByTagAndChangeVector().ExecuteAsync(store.Instance, token: cts.Token);
+                    await new ActorsByChangeVector().ExecuteAsync(store.Instance, token: cts.Token);
+
+                    return new Status.Success(NotUsed.Instance);
+                }
+                catch (DatabaseDoesNotExistException e)
+                {
+                    Thread.Sleep(TimeSpan.FromMilliseconds(250));
+                }
+                catch (DatabaseDisabledException e)
+                {
+                    //database locked, try again
+                    Thread.Sleep(TimeSpan.FromMilliseconds(250));
```

Comment:
I think we need to call it once out side the loop

Reply:
done